be/src/main/java/yeonba/be/user/repository/UserQuery.java

```diff
@@ -0,0 +1,21 @@
+package yeonba.be.user.repository;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import yeonba.be.exception.ExceptionType;
+import yeonba.be.exception.GeneralException;
+import yeonba.be.user.entity.User;
+
+@Component
+@RequiredArgsConstructor
+public class UserQuery {
+
+  private final UserRepository userRepository;
+
+  public User findById(long userId) {
+
+    return userRepository.findById(userId)
+        .orElseThrow(() -> new GeneralException(ExceptionType.USER_NOT_FOUND));
```

Comment:
제가 잘 몰라서 그러는데 `GeneralException`는 언제 사용하는 것인지 알려주시기 바랍니다. 지금과 같은 상황에 사용해야 하는 게 아니라면 일단 `IllegalArgumentException`을 사용하기로 했던 것같은데 어떤가요?
```suggestion
        .orElseThrow(() -> new IllegalArgumentException(ExceptionType.USER_NOT_FOUND));
```

Reply:
`GeneralException`은 서비스 고유 예외를 나타내며 `ExceptionAdvice`에서 해당 예외를 전담해 핸들링하는 로직이 따로 존재합니다. 따라서 서비스 고유 예외가 발생하는 부분을 일관성 있게 전부 `GeneralException`을 활용하였습니다.

Reply:
하지만 현재 `ExceptionAdvice`에서 일부 `HttpMessageNotReadableException`을 의도한 대로 커스텀 처리하지 못하는 문제가 발생하여 예외 처리 로직을 수정할 계획입니다. 따라서
- 일단 `IllegalArgumentException`으로 예외 발생 부분을 전부 작성
- `GeneralException` 핸들러 로직은 정상 작동하는데, 제가 작성한 방식대로 구성
두 방향 중 어느 쪽이 적절한 지 의견 나누고 정하는 것이 좋을 것 같습니다.